src/mutex.cc

```diff
@@ -21,27 +21,47 @@ namespace toit {
 void Locker::leave() {
   Thread* thread = Thread::current();
   if (thread->locker_ != this) FATAL("unlocking would break lock order");
-  thread->locker_ = previous_;
-  // Perform the actual unlock.
-  mutex_->unlock();
+  thread->locker_ = previous();
```

Comment:
I'm considering setting `previous_` to `NULL` here, just because it feels weird that we keep the state in the locker and only get rid of it when we ever re-enter.

Reply:
setting to NULL works for me.